### PR TITLE
[DomCrawler] minor performance improvement

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? |no 
| Tests pass?   | yes    
| Fixed tickets | n/a   
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Review by Simperfit — APPROVED** [review]

**Comment by ostrolucky** [comment]
Changing weak to strict comparison in public method is potentially breaking change and should not be done in patch release

**Comment by dunglas** [comment]
@ostrolucky not here, because the test is in a `foreach` (https://3v4l.org/ElUI2) and the parameter is documented as `int` in the PHPDoc.

**Comment by ostrolucky** [comment]
I know `$i` is guaranteed to be integer, but not `$position`. Yes, blame can be put on user if they didn't make sure argument is integer. That's why I consider this change ok to do in minor release without migration path. For patch release IMHO it is too risky to change how code processes arguments with no good reason, regardless if it's documented in phpdoc. Few years back double equals operator was more common and applications could count with the fact that `int` in API meant more like `number`.

Also, it seems this provides no performance boost if data types compared are same https://stackoverflow.com/a/14520998/524965

**Comment by javiereguiluz** [comment]
I agree with @ostrolucky about the strict comparison: this kind of changes is too risky in a patch version. Better do that in master. Thanks!

**Comment by jakzal** [comment]
How big is the performance gain?

**Comment by dunglas** [comment]
`$position` is documented as int in the Docblock, we don't garantee BC on unsupported types, but I'll target master as it's not a bug fix.

@jakzal I've not benchmarked it, but on large documents it should be important (calling a method in a loop for nothing always come with a big cost).

**Comment by fabpot** [comment]
Looks like there are some unrelated commits here.

**Comment by dunglas** [comment]
@fabpot Fixed.

@ostrolucky @javiereguiluz the comparison has already been removed in master, problem fixed!

**Review by javiereguiluz — APPROVED** [review]

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -1081,8 +1081,9 @@ protected function sibling($node, $siblingDir = 'nextSibling')
     {
         $nodes = array();
 
+        $currentNode = $this->getNode(0);
         do {
-            if ($node !== $this->getNode(0) && 1 === $node->nodeType) {
+            if ($node !== $currentNode && 1 === $node->nodeType) {
```
[the remark this answers is not in the document]
Comment (dunglas) [reply src/Symfony/Component/DomCrawler/Crawler.php]: I agree, can I do it in the same PR?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @dunglas.

**Pull request closed without merging**